### PR TITLE
chore(eslint): improve eslint; remove uneeded dependencies

### PR DESCRIPTION
**Pull request by diehbria**
## Overview
Improve consistency of linting standards and remove unused packages, as well as unnecessary deviations in style

- remove unused linting packages in scene composer
- consolidate lint configs on table and related table
- fix regression on typescript types on `dashboard`
- fix regression where typescript types were not enforced on tests
- enforce env variables being specified in the turborepo pipeline. This is important for caching of packages.
- remove env variables used for mocks within dashboard package
- applied `eslint fix`
## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/rollup.config.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ export default [
       json(),
       typescript({
         tsconfig: './tsconfig.json',
+        exclude: ['**/*.test.ts', '**/*.test.tsx', '**/*.spec.ts', '**/*.spec.tsx'],
```
Comment (square-li) [inline packages/dashboard/rollup.config.js:43]: it should be 
``` 
typescript({
  tsconfig: './tsconfig.json',
  tsconfigOverride: {
    exclude: ['**/*.test.ts', '**/*.test.tsx', '**/*.spec.ts', '**/*.spec.tsx'],
  },
}),
```

**Review by square-li — CHANGES_REQUESTED** [review]

File: packages/dashboard/rollup.config.js (hunk 1 above)
Comment (square-li) [reply packages/dashboard/rollup.config.js:43]: https://github.com/awslabs/iot-app-kit/commit/49510bacc6f5c385cd3d153aa0a127b5b94dfc70#diff-8d2edfca7863606e1df4d8d8da1e7150063abfa13c9a2bfe654f9fbeacc1a60cL41-L43

Comment (diehbria) [reply packages/dashboard/rollup.config.js]: fixed :-)

**Review by square-li — DISMISSED** [review]

**Review by sheilaXu — DISMISSED** [review]

File: packages/related-table/package.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
     "dev": "npm-watch build",
     "generate-barrels": "barrelsby --config ./barrel-config.json",
     "test": "jest",
-    "lint": "eslint . --max-warnings=27",
+    "lint": "eslint . --max-warnings=54",
```
Comment (diehbria) [inline packages/related-table/package.json:43]: there were warnings present but obfuscated by incorrect settings

**Review by square-li — APPROVED** [review]

**Pull request merged**